### ISSUE TITLE
SendSelectionAsync: Cannot use message from different user

### ISSUE THREAD

**Issue by stageosu** [Playwo/Discord.InteractivityAddon#16]
I'm not sure why I'm getting this error as I'm trying to modify a message the bot is sending. On my development bot, it works just fine, but when I am using it in production, I'm getting this error.

Stack trace:
```
[02-13-2021 04:19:44:804] fail: Kaguya.Workers.DiscordWorker[0]
      Exception encountered when executing command. Message:
      Discord.Commands.CommandException: Error occurred executing "SearchCommand" for Stage#0001 in Kaguya Testing/general.
       ---> System.ArgumentException: Cannot use message from different user!
         at Interactivity.InteractivityService.SendSelectionAsync[TValue](BaseReactionSelection`1 selection, IMessageChannel channel, Nullable`1 timeout, IUserMessage message, Nullable`1 runOnGateway, CancellationToken cancellationToken)
         at Kaguya.Discord.Commands.Music.Search.SearchCommand(String search) in /Kaguya/Discord/Commands/Music/Search.cs:line 114
         at Discord.Commands.ModuleClassBuilder.<>c__DisplayClass6_0.<<BuildCommand>g__ExecuteCallback|0>d.MoveNext()
      --- End of stack trace from previous location ---
         at Discord.Commands.CommandInfo.ExecuteInternalAsync(ICommandContext context, Object[] args, IServiceProvider services)
         --- End of inner exception stack trace ---

```

Here is my code, with line numbers. Line 114 is where the exception is thrown.
```cs
95 var embed = new KaguyaEmbedBuilder(Color.Blue)
96             .WithTitle("Kaguya Music: Search Result")
97             .WithDescription($"{Context.User.Mention} {line1}")
98             .WithFields(embedFields);
99 
100 var toModify = await SendEmbedAsync(embed);
101 var builder = new KaguyaReactionSelectionBuilder<int>()
102                 .WithSelectables(new Dictionary<IEmote, int>
103                 {
104                    { _commonEmotes.EmojisOneThroughFive[0], 0 },
105                    { _commonEmotes.EmojisOneThroughFive[1], 1 },
106                    { _commonEmotes.EmojisOneThroughFive[2], 2 },
107                    { _commonEmotes.EmojisOneThroughFive[3], 3 },
108                    { _commonEmotes.EmojisOneThroughFive[4], 4 }
109                })
110                .WithEnableDefaultSelectionDescription(false)
111                .WithDeletion(DeletionOptions.AfterCapturedContext | DeletionOptions.Invalids)
112                .WithUsers(Context.User);
113
114 var result = await _interactivityService.SendSelectionAsync(builder.Build(), Context.Channel, 
115 TimeSpan.FromMinutes(2), toModify);
```

**Comment by stageosu** [comment]
Actually, I think I have a solution. I'll let you know how the implementation goes (it worked fine on test bot, need to test on remote).

**Comment by Playwo** [comment]
I have this piece of code in the part that actually sends the Selection: 
```cs
if (message != null)
{
    if (message.Author != Client.CurrentUser)
    {
        throw new ArgumentException("Cannot use message from different user!");
    }

    await selection.InternalModifyMessageAsync(message).ConfigureAwait(false); ;
}
```

That is because most features don't really work with other people's messages (Mostly changing the message content before and after the selection starts)
But it does not have to make sense for other implementations of Selection. So I think I should move that exception to the default implementation of Selection instead of having it in the service itself.

**Comment by Playwo** [comment]
Ok it will stay in the service but I added a type check so that it does not get checked for custom implementations. The reason for that is because I did not want to add the client as parameter for ModifyMessageAsync. It's not really necessary to have the client in the Selection so I preferred to do it this way.

**Comment by stageosu** [comment]
> Ok it will stay in the service but I added a type check so that it does not get checked for custom implementations. The reason for that is because I did not want to add the client as parameter for ModifyMessageAsync. It's not really necessary to have the client in the Selection so I preferred to do it this way.

That should solve it, and I will say I completely understand why that is in the code it was just weird getting the error. I'll leave this issue open because I might be in the wrong myself with my implementation. I have a pending update where I change my code here to be a lot more "proper" with how you probably expected developers to set up this part of your lib. Either way I'll use this new version, I'll keep you updated. Appreciate the solution.

**Comment by stageosu** [comment]
Here is the fix that worked for me on 2.4.0-dev.20210130.1

```cs
var embed = new KaguyaEmbedBuilder(Color.Blue)
            .WithTitle("Kaguya Music: Search Result")
            .WithDescription($"{Context.User.Mention} {line1}")
            .WithFields(embedFields);

var builder = new KaguyaReactionSelectionBuilder<int>()
              .WithSelectables(new Dictionary<IEmote, int>
              {
                  { _commonEmotes.EmojisOneThroughFive[0], 0 },
                  { _commonEmotes.EmojisOneThroughFive[1], 1 },
                  { _commonEmotes.EmojisOneThroughFive[2], 2 },
                  { _commonEmotes.EmojisOneThroughFive[3], 3 },
                  { _commonEmotes.EmojisOneThroughFive[4], 4 }
              })
              .WithDeletion(DeletionOptions.AfterCapturedContext | DeletionOptions.Invalids)
              .WithEnableDefaultSelectionDescription(false)
              .WithSelectionEmbed(PageBuilder.FromEmbedBuilder(embed))
              .WithUsers(Context.User);

var result = await _interactivityService.SendSelectionAsync(builder.Build(), Context.Channel, TimeSpan.FromMinutes(2));
```

**Comment by Playwo** [comment]
I'm pretty sure that this has been fixed, reopen if I'm wrong
